src/shared/db.ts

```diff
@@ -33,3 +35,19 @@ function parseNextAuthUser(loginUser: NextUser): Prisma.UserCreateInput {
     avatar: loginUser.image ?? undefined,
   };
 }
+
+export async function fetchUserWithSession(session: Session | null) {
+  if (!session?.user?.email) {
+    const origin = getOriginPath();
+    return redirect(makeRedirectURL(origin));
+  }
+
+  const loggedUser = await fetchUserBy({ email: session.user.email });
+
+  if (!loggedUser) {
+    const origin = getOriginPath();
+    return redirect(makeRedirectURL(origin));
+  }
+
+  return loggedUser;
+}
```

Comment:
to avoid repeating yourself

```suggestion
export async function fetchUserWithSession(session: Session | null) {
  const redirectUrl = makeRedirectURL(getOriginPath())
  if (!session?.user?.email) return redirect(redirectUrl);

  const loggedUser = await fetchUserBy({ email: session.user.email });
  if (!loggedUser) return redirect(redirectUrl);

  return loggedUser;
}
```

Reply:
Done!